packages/@glimmerx/babel-plugin-component-templates/index.js

```diff
@@ -107,54 +109,67 @@ function getTemplateTokens(html, nativeTokens) {
 module.exports = function(babel, options) {
   const { types: t, parse } = babel;
   const {
+    // Whether or not we'ret targeting Ember. If so, we use the Ember global
+    // instead of adding imports, and we use different default import paths.
+    ember = false,
+
     setTemplatePath = '@glimmer/core',
-    setTemplateName = 'setComponentTemplate',
+    setTemplateName = ember ? '_setComponentTemplate' : 'setComponentTemplate',
+
     createTemplatePath = '@glimmer/core',
     createTemplateName = 'createTemplate',
+
     templateOnlyComponentPath = '@glimmer/core',
-    templateOnlyComponentName = 'templateOnlyComponent',
-    precompile: precompileOptions,
+    templateOnlyComponentName = ember ? '_templateOnlyComponent' : 'templateOnlyComponent',
+
+    precompile: precompileOptions = {},
+
+    // Users can pass a custom precompile function. This is used by Ember.
+    precompileTemplate = glimmerPrecompileTemplate,
   } = options || {};
 
-  const shouldPrecompile = !(precompileOptions && precompileOptions.disabled);
+  const shouldPrecompile = precompileOptions.disabled !== true;
 
-  function maybeAddSetTemplateImport(state, programPath) {
-    if (!state.setTemplateId) {
-      state.setTemplateId = addNamed(programPath, setTemplateName, setTemplatePath, {
-        importedType: 'es6',
-      }).name;
+  function maybeAddImport(state, programPath, name, importPath) {
+    let stateKey = `__import__${name}`;
+
+    if (!state[stateKey]) {
+      if (ember) {
+        // In Ember, we destructure the value from the Ember global
+        const globalImport = t.variableDeclaration('const', [
+          t.variableDeclarator(
+            t.objectPattern([t.objectProperty(t.identifier(name), t.identifier(name))]),
+            t.identifier('Ember')
+          ),
+        ]);
```

Comment:
it makes the rest of the code pretty similar, we just call the identifier directly rather than having to add a chain node. We would be spreading a lot more Ember related code around the transform if we did it that way.